zmq/examples/threaded_loopback.ml

```diff
@@ -30,6 +30,6 @@ let () =
     Thread.create subscription subscribe_socket in
 
   List.iter Thread.join  [subscribe_thread; publish_thread];
-  List.iter Socket.close [subscribe_socket; publish_socket];
+  List.iter Socket.close [(subscribe_socket :> [`Sub | `Pub] Socket.t); (publish_socket :> [`Sub | `Pub] Socket.t)];
```

Comment:
Yep that is the easiest option though I thought that it may be useful though to demonstrate the subtyping approach (since I personally didn't know about it before working on this stuff).

Reply:
Sounds reasonable.